grpc/src/main/java/com/netflix/conductor/grpc/AbstractProtoMapper.java

```diff
@@ -0,0 +1,1144 @@
+package com.netflix.conductor.grpc;
+
+import com.google.protobuf.Value;
+import com.netflix.conductor.common.metadata.events.EventExecution;
+import com.netflix.conductor.common.metadata.events.EventHandler;
+import com.netflix.conductor.common.metadata.tasks.PollData;
+import com.netflix.conductor.common.metadata.tasks.Task;
+import com.netflix.conductor.common.metadata.tasks.TaskDef;
+import com.netflix.conductor.common.metadata.tasks.TaskExecLog;
+import com.netflix.conductor.common.metadata.tasks.TaskResult;
+import com.netflix.conductor.common.metadata.workflow.DynamicForkJoinTask;
+import com.netflix.conductor.common.metadata.workflow.DynamicForkJoinTaskList;
+import com.netflix.conductor.common.metadata.workflow.RerunWorkflowRequest;
+import com.netflix.conductor.common.metadata.workflow.SkipTaskRequest;
+import com.netflix.conductor.common.metadata.workflow.StartWorkflowRequest;
+import com.netflix.conductor.common.metadata.workflow.SubWorkflowParams;
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import com.netflix.conductor.common.run.TaskSummary;
+import com.netflix.conductor.common.run.Workflow;
+import com.netflix.conductor.common.run.WorkflowSummary;
+import com.netflix.conductor.proto.DynamicForkJoinTaskListPb;
+import com.netflix.conductor.proto.DynamicForkJoinTaskPb;
+import com.netflix.conductor.proto.EventExecutionPb;
+import com.netflix.conductor.proto.EventHandlerPb;
+import com.netflix.conductor.proto.PollDataPb;
+import com.netflix.conductor.proto.RerunWorkflowRequestPb;
+import com.netflix.conductor.proto.SkipTaskRequestPb;
+import com.netflix.conductor.proto.StartWorkflowRequestPb;
+import com.netflix.conductor.proto.SubWorkflowParamsPb;
+import com.netflix.conductor.proto.TaskDefPb;
+import com.netflix.conductor.proto.TaskExecLogPb;
+import com.netflix.conductor.proto.TaskPb;
+import com.netflix.conductor.proto.TaskResultPb;
+import com.netflix.conductor.proto.TaskSummaryPb;
+import com.netflix.conductor.proto.WorkflowDefPb;
+import com.netflix.conductor.proto.WorkflowPb;
+import com.netflix.conductor.proto.WorkflowSummaryPb;
+import com.netflix.conductor.proto.WorkflowTaskPb;
+import java.lang.IllegalArgumentException;
+import java.lang.Object;
+import java.lang.String;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Generated;
+
+@Generated("com.netflix.conductor.protogen.ProtoGen")
+public abstract class AbstractProtoMapper {
+    public EventExecutionPb.EventExecution toProto(EventExecution from) {
+        EventExecutionPb.EventExecution.Builder to = EventExecutionPb.EventExecution.newBuilder();
+        if (from.getId() != null) {
```

Comment:
why is this null check needed? to simplify generator logic?

Reply:
It is not safe to call a setter on a ProtoBuf object builder with a `null` argument, so we have to explicitly check for all nullable types before we do the mapping.

Reply:
Makes sense.